src/core/query.ts

```diff
@@ -13,6 +13,18 @@ import {
 } from './utils'
 import { makeQueryInstance } from './queryInstance'
 
+export interface QueryState<T, TError = Error> {
+  data?: T
+  error?: TError | null
+  failureCount: number
+  isFetching: boolean
+  canFetchMore?: boolean
+  isStale: boolean
+  status: 'idle' | 'loading' | 'error' | 'success'
```

Comment:
Instead of inlining, we should probably make a explicit `type` for `status` which will be easier for consumers and can potentially be reused across.

Reply:
Yes. Make all the things reusable. In fact you’ll see soon that status is an enum that has constants in the utils file.